### PR TITLE
feat: basic introduction of plugin-chart-ol

### PR DESCRIPTION
**Pull request by jansule**
### SUMMARY

This sets up the basic introduction of the plugin-chart-ol. This also fixes the test setup and does some minor but needed adjustments to other packages included in superset.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/jest.config.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ module.exports = {
     // mapping plugins of superset-ui to souce code
     '@superset-ui/(.*)$': '<rootDir>/node_modules/@superset-ui/$1/src',
   },
```
Comment (JakobMiksch) [inline superset-frontend/jest.config.js:31]: a minor comment would be useful for explanation

File: superset-frontend/plugins/plugin-chart-ol/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# superset-ol-plugin
+
+An OpenLayers map that displays charts for single features. Superset Chart Plugin.
+
+### Usage
```
Comment (JakobMiksch) [inline superset-frontend/plugins/plugin-chart-ol/README.md:5]:
```suggestion
## Usage
```

File: superset-frontend/plugins/plugin-chart-ol/src/components/styled/StyledControlFormItem.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// import ControlFormItem from '@superset-ui/chart-controls/components/ControlForm/ControlFormItem';
```
Comment (JakobMiksch) [inline superset-frontend/plugins/plugin-chart-ol/src/components/styled/StyledControlFormItem.tsx:20]: remove commented code

File: superset-frontend/src/visualizations/presets/MainPreset.js [hunk 4]
```diff
@@ -166,6 +167,17 @@ export default class MainPreset extends Preset {
         new TimeGrainFilterPlugin().configure({ key: 'filter_timegrain' }),
         new EchartsTreeChartPlugin().configure({ key: 'tree_chart' }),
         new HandlebarsChartPlugin().configure({ key: 'handlebars' }),
+        new SupersetOlPlugin({
+          defaultLayers: [
```
Comment (JakobMiksch) [inline superset-frontend/src/visualizations/presets/MainPreset.js:171]: maybe we can use a different default layer here that is more generic. for example the standard OSM tile

File: superset-frontend/jest.config.js (hunk 1 above)
Comment (jansule) [reply superset-frontend/jest.config.js:31]: This is a default configuration when using es6 modules with jest.

File: superset-frontend/src/visualizations/presets/MainPreset.js (hunk 4 above)
Comment (jansule) [reply superset-frontend/src/visualizations/presets/MainPreset.js:171]: Sure. I will add a TODO

**Pull request merged**